service/http.proxy/src/main/java/org/kaazing/gateway/service/http/proxy/HttpProxyService.java

```diff
@@ -73,4 +86,37 @@ private void checkForTrailingSlashes(ServiceContext serviceContext) {
     protected HttpProxyServiceHandler createHandler() {
         return new HttpProxyServiceHandler();
     }
+
+    /**
+     * Method returning true if overriding accept/connect options found
+     * @param serviceContext
+     * @return
+     */
+    private boolean noOverridingAcceptConnectOptions(ServiceContext serviceContext) {
+        AcceptOptionsContext acceptOptionsContext = serviceContext.getAcceptOptionsContext();
+        ConnectOptionsContext connectOptionsContext = serviceContext.getConnectOptionsContext();
+
+        boolean bindsEmpty = acceptOptionsContext.getBinds().isEmpty();
+        boolean hasAcceptTransport = hasTransport(acceptOptionsContext.asOptionsMap());
+        boolean hasConnectTransport = hasTransport(connectOptionsContext.asOptionsMap());
+
+        return bindsEmpty && !hasAcceptTransport && !hasConnectTransport;
+    }
+
+    /**
+     * Method returning whether option list has *.transport(s)
+     * @param map
+     * @return
+     */
+    private boolean hasTransport(Map<String, Object> map) {
```

Comment:
This method assumes knowledge of the fact that transport accept options are in *.transport format (which is actually an information to be encapsulated within DefaultAcceptOptionsContext/DefaultConnectOptionsContext).

Should we rather add a method like *hasTransports* in AcceptOptionsContex/ConnectOptionsContext interfaces?